src/store/transactions/useTransactionPoller.ts

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useEffect } from 'react';
+import { transactionsSlice, useTransactions } from './reducer';
+import { useWeb3React } from '@web3-react/core';
+import { useDispatch } from 'react-redux';
+import { TxnStatus } from './types';
+
+const TX_POLL_INTERVAL = 1000 * 5; // 5 seconds
+
+const useTransactionPoller = () => {
+  const { savedTransactions } = useTransactions();
+  const { library } = useWeb3React();
+  const dispatch = useDispatch();
+
+  const checkTransactionStatuses = useCallback(async () => {
+    savedTransactions.forEach((txn) => {
+      library?.getTransactionReceipt(txn.hash).then((receipt: any) => {
+        if (receipt) {
+          dispatch(
+            transactionsSlice.actions.setTransactionStatus({
+              ...txn,
+              status: receipt.status === 1 ? TxnStatus.Mined : TxnStatus.Failed,
+            })
+          );
+        } else {
+          console.log('no reciept?');
+          // pending on chain?
+        }
+      });
+    });
+  }, [savedTransactions]);
+
+  useEffect(() => {
+    // @ts-ignore
+    let interval;
```

Comment:
Could you remove` @ts-ignore` by adding the following:
```suggestion
    let interval: number;
```